### PR TITLE
chore: Add publish to Hub CI workflow

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloud/issues/303 (internal issue).

### CODE REVIEW

**Comment by erezrokah** [comment]
Waiting for https://github.com/cloudquery/cloudquery/pull/13880 to be released

File: .github/workflows/publish_plugin_to_hub.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+              owner: "cloudquery",
+              repo: context.repo.repo,
+              tag: context.ref.replace('refs/tags/', ''),
+            });
+            return data.body;
+
+      - name: Run publish package command
+        working-directory: ${{ steps.split.outputs.plugin_dir }}
+        run: |
+          go run main.go package -m "${{ steps.release-notes.outputs.result }}" . ${{ steps.split.outputs.plugin_version }}
```
Comment (hermanschaaf) [inline .github/workflows/publish_plugin_to_hub.yml:65]: Right now this will only work for Go plugins, are other plugins being excluded somehow?

Comment (erezrokah) [reply .github/workflows/publish_plugin_to_hub.yml:65]: Ah right, I'll update it to ignore other plugins

Comment (erezrokah) [reply .github/workflows/publish_plugin_to_hub.yml]: Fixed and based on the logic from https://github.com/cloudquery/cloudquery/blob/7d3aaa8950383cbaf5de5a016c9cd594bbbbc054/.github/workflows/release_plugin.yml#L48

I intentionally duplicated the file so not to interfere with existing publishing flow. We can merge them later on if needed

**Review by hermanschaaf — APPROVED** [review]
Nice 🚀

**Pull request merged**